syft/execution/state.py

```diff
@@ -64,7 +64,9 @@ def read(self):
                 # for all the placeholders in this sub plan, we report a copy of them
                 # in the parent plan and notify their origin using the #inner tag
                 for placeholder in self.state_placeholders:
+                    ex_id = placeholder.id
                     placeholder = placeholder.copy()
+                    placeholder.id = ex_id
```

Comment:
Does `copy` change the ID of the original `Placeholder`? Weird!

Reply:
It does, maybe because we don't want to have 2 objects with the same id on the same worker

Reply:
Ah, right.